# S03 T01: Estructura de dades

### Exercici 1

##### Lista que agrupa los meses del año en trimestres

In [159]:
mesesdelaño = [["enero", "febrero", "marzo"],["abril", "mayo", "junio"],
["julio", "agosto", "septiembre"],["octubre", "noviembre", "diciembre"]]

### Exercici 2

#### Acceder al segundo mes del primer trimestre

In [169]:
mesesdelaño[0][1]

'febrero'

#### Acceder a los meses del primer trimestre

In [161]:
mesesdelaño[0]

['enero', 'febrero', 'marzo']

#### Acceder a septiembre y octubre

In [162]:
mesesdelaño[2][2]

'septiembre'

In [163]:
mesesdelaño[3][0]

'octubre'

### Exercici 3

#### Crear lista con números desordenados

In [170]:
numeros = [3, 5, 9, 2, 15, 4, 19, 345, 201, 15, 8, 1983, 4]

#### ¿Cuántos números tiene?

In [165]:
print(len(numeros))

13


La lista tiene 13 números

#### ¿Cuántas veces aparece el número 3?

In [166]:
numeros.count(3)

1

El número 3 aparece una vez

#### ¿Cuántas veces aparecen los números 3 y 4?

In [171]:
import collections
from collections import Counter
Counter(numeros)

Counter({3: 1,
         5: 1,
         9: 1,
         2: 1,
         15: 2,
         4: 2,
         19: 1,
         345: 1,
         201: 1,
         8: 1,
         1983: 1})

El número 3 aparece una vez y el 4 dos veces

#### ¿Cuál es el número más grande de la lista?

In [172]:
numeros.sort(reverse = True)
print(numeros[0])

1983


El número más grande de la lista es el 1983

#### ¿Cuáles son los tres números más pequeños de la lista?

In [173]:
numeros.sort()
print(numeros[0:3])

[2, 3, 4]


Los tres números más pequeños de la lista son el 2, el 3 y el 4

#### ¿Cuál es el rango de la lista?

In [174]:
range(1)
print(numeros)

[2, 3, 4, 4, 5, 8, 9, 15, 15, 19, 201, 345, 1983]


### Exercici 4

#### Lista de la compra

In [175]:
compra = { "Pomes" : {"Cantidad" : 5, "precio" : 0.42}, 
          "Peres" : {"Cantidad" : 3, "precio" : 0.66} }

#### Agregar una fruta a la lista

In [176]:
compra = { "Pomes" : {"Cantidad": 5, "Precio": 0.42}, 
          "Peres" : {"Cantidad": 3, "Precio": 0.66} }
compra.update({"Melocoton" : {"Cantidad": 3, "Precio": 0.50}})

#### ¿Cuánto han costado las peras en total?

In [177]:
print(compra.get('Peres'))

{'Cantidad': 3, 'Precio': 0.66}


In [178]:
print(3*0.66) 

1.98


Las peras en total han costado €1.98

#### ¿Cuántas frutas hemos comprado en total?

In [185]:
for x in compra:
    print(compra[x])

{'Cantidad': 5, 'Precio': 0.42}
{'Cantidad': 3, 'Precio': 0.66}
{'Cantidad': 3, 'Precio': 0.5}


In [186]:
print(5+3+3)

11


En total hemos comprado 11 frutas

#### ¿Cuál es la fruta más cara?

In [181]:
for x, y in compra.items():
    print(x, y)

Pomes {'Cantidad': 5, 'Precio': 0.42}
Peres {'Cantidad': 3, 'Precio': 0.66}
Melocoton {'Cantidad': 3, 'Precio': 0.5}


La fruta más cara son las Peres